common-components/proto/api/type/money.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package api.type;
+
+option cc_enable_arenas = true;
+option go_package = "aip.dev/type/money;money";
+option java_multiple_files = true;
+option java_outer_classname = "MoneyProto";
+option java_package = "dev.aip.type";
+option objc_class_prefix = "AIPTYPE";
+
+import "api/type/decimal.proto";
+
+// Represents an amount of money with its currency type.
+message Money {
+  // A currency code.
+  //
+  // This may be a three-letter currency code defined in ISO 4217.
+  //
+  // APIs may define additional currency codes that are not included in the ISO
+  // 4217 standard (for example, virtual currencies or cryptocurrencies). These
+  // must start with a `.` character, in order to distinguish them from
+  // potential future ISO 4217 codes.
+  //
+  // Examples:
+  //   "USD"  - ISO 4217 code for United States dollar
+  //   ".BTC" - Potential API-defined extension for Bitcoin.
+  //   ".RBX" - Potential API-defined extension for the virtual currency Robux.
+  string currency_code = 1;
+
+  // The amount of currency.
+  Decimal amount = 2;
```

Comment:
Value? (No strong opinion.)

Reply:
I don't feel strongly about this either, but here's my rationale for `amount` over `value`:

Weak precedent:  google.type.Money [refers](https://github.com/googleapis/googleapis/blob/master/google/type/money.proto#L31) to units and nanos as making up the "amount".

Also, the whole message is the value, right?  `value` is used in e.g. google/protobuf/wrappers.proto, where that field really does represent the entire value of the message.

I could see an argument for `quantity` over `amount`...but `value` feels too generic to me.

Reply:
I'm convinced of `amount`, I think.